readthedocs/projects/models.py

```diff
@@ -1401,6 +1402,60 @@ class EnvironmentVariable(TimeStampedModel, models.Model):
     def __str__(self):
         return self.name
 
+    def _get_last_built_version(self):
+        """Returns the version that was last built."""
+        version = self.project.versions.all().order_by('-builds__date').first()
+        return version
+
+    def _get_env_json_paths(self, version):
+        """
+        Returns the list containing paths to ``readthedocs-environment.json`` file.
+
+        It returns paths for both python and conda environment.
+        """
+        paths = [
+            os.path.join(
+                self.project.doc_path,
+                'conda',  # if conda env
+                version.slug,
+                'readthedocs-environment.json'
+            ),
+            os.path.join(
+                self.project.doc_path,
+                'envs',  # if python env
+                version.slug,
+                'readthedocs-environment.json'
+            ),
+        ]
+
+        return paths
+
+    def update_env_json(self):
```

Comment:
Did you add these methods here for any particular reason? Why not just use these helpers from https://github.com/rtfd/readthedocs.org/blob/a4e9623621b7618f7376c665768b6ae914b38e3d/readthedocs/doc_builder/python_environments.py#L134

I think it should be easier to add just one more check to https://github.com/rtfd/readthedocs.org/blob/a4e9623621b7618f7376c665768b6ae914b38e3d/readthedocs/doc_builder/python_environments.py#L142

Reply:
@humitos 
I have updated the PR.